src/modules/tools/zprobe/ZProbe.cpp

```diff
@@ -336,9 +336,16 @@ void ZProbe::on_gcode_received(void *argument)
             return;
         }
 
-        if(this->pin.get()) {
-            gcode->stream->printf("error:ZProbe triggered before move, aborting command.\n");
-            return;
+        if(gcode->subcode == 4 || gcode->subcode == 5) {
```

Comment:
This code is way too verbose and can be significantly simplified.

Reply:
eg 
```
if(((gcode->subcode == 4 || gcode->subcode == 5) && !this->pin.get()) || ((gcode->subcode == 2 || gcode->subcode == 3) && this->pin.get())) {
  gcode->stream->printf("error:ZProbe triggered before move, aborting command.\n");
}

or even better....

if(this->pin.get() ^ (gcode->subcode >= 4)) {
￼             gcode->stream->printf("error:ZProbe triggered before move, aborting command.\n");
￼             return;
}
```

Reply:
I like the 'even better' code.  very smart and takes advantage of gcode.subcode being already constrained to be 2-5.